wormhole-connect/src/utils/wallet/solana.ts

```diff
@@ -69,52 +75,129 @@ export function fetchOptions() {
   };
 }
 
+// This function signs and sends the transaction while constantly checking for confirmation
+// and resending the transaction if it hasn't been confirmed after the specified interval
+// NOTE: The caller is responsible for simulating the transaction and setting any compute budget
+// or priority fee before calling this function
+// See https://docs.triton.one/chains/solana/sending-txs for more information
 export async function signAndSendTransaction(
   request: SolanaUnsignedTransaction<Network>,
   wallet: Wallet | undefined,
   options?: ConfirmOptions,
 ) {
-  if (!wallet || !wallet.signAndSendTransaction) {
-    throw new Error('wallet.signAndSendTransaction is undefined');
+  if (!wallet) throw new Error('Wallet not found');
+  if (!config.rpcs.solana) throw new Error('Solana RPC not found');
+
+  const commitment = options?.commitment ?? 'finalized';
+
+  const unsignedTx = request.transaction.transaction;
```

Comment:
I assume this property chain has checked before undefined?

Reply:
This property will exist on the `SolanaUnsignedTransaction` - passing that is the caller's responsibility